Neos.EventSourcedContentRepository/Classes/Domain/Context/NodeAggregate/NodeAggregateCommandHandler.php

```diff
@@ -97,6 +106,281 @@ public function __construct(
+     * @throws DimensionSpacePointNotFound If the given dimension space point is not in the allowed dimension space
+     * @throws NodeConstraintException If a node aggregate of that type is not allowed to be created as a descendant of its parents
+     * @throws NodeNameIsAlreadyOccupied If the given node name is already taken in any of the dimension space points the node will be visible in
+     * @throws NodeTypeNotFoundException If the given type does not exist
+     * @throws \Neos\Flow\Property\Exception
+     * @throws \Neos\Flow\Security\Exception
+     * @throws \Neos\EventSourcedContentRepository\Domain\Context\Node\NodeAggregatesTypeIsAmbiguous
+     */
+    public function handleCreateNodeAggregateWithNode(CreateNodeAggregateWithNode $command): CommandResult
+    {
+        $this->readSideMemoryCacheManager->disableCache();
+
+        $nodeAggregate = $this->getNodeAggregate($command->getContentStreamIdentifier(), $command->getNodeAggregateIdentifier());
+        $this->requireDimensionSpacePointToExist($command->getOriginDimensionSpacePoint());
+        $nodeType = $this->getNodeType($command->getNodeTypeName());
+        $this->requireAutoCreatedChildNodeTypesToExist($nodeType);
+        $this->requireConstraintsImposedByAncestorsAreMet($command->getContentStreamIdentifier(), $nodeType, $command->getNodeName(), [$command->getParentNodeAggregateIdentifier()]);
+        #$this->requireNodeAggregateToCurrentlyNotExist($command->getContentStreamIdentifier(), $command->getNodeAggregateIdentifier());
+        #$this->requireNodeAggregateToCurrentlyExist($command->getContentStreamIdentifier(), $command->getParentNodeAggregateIdentifier());
+        if ($nodeType->isOfType(NodeTypeName::ROOT_NODE_TYPE_NAME)) {
+            throw new NodeTypeIsOfTypeRoot('Node type "' . $nodeType . '" for non-root node "' . $command->getNodeAggregateIdentifier() . '" is of type root.', 1541765806);
+        }
+
+        if ($command->getSucceedingSiblingNodeAggregateIdentifier()) {
+            $this->requireNodeAggregateToCurrentlyExist($command->getContentStreamIdentifier(), $command->getSucceedingSiblingNodeAggregateIdentifier());
+        }
+        $this->requireNodeAggregateToBeVisibleInDimensionSpacePoint($command->getContentStreamIdentifier(), $command->getParentNodeAggregateIdentifier(), $command->getOriginDimensionSpacePoint());
+
+        $specializations = $this->interDimensionalVariationGraph->getSpecializationSet($command->getOriginDimensionSpacePoint());
+        // $parentAggregate = $this->getNodeAggregate($command->getContentStreamIdentifier(), $command->getParentNodeAggregateIdentifier());
+        $projectedParentAggregate = $this->contentGraph->findNodeAggregateByIdentifier($command->getContentStreamIdentifier(), $command->getParentNodeAggregateIdentifier());
+        $visibleDimensionSpacePoints = $specializations; # $specializations->intersect($projectedParentAggregate->getVisibleInDimensionSpacePoints());
```

Comment:
can you add some comments and TODOs to the code above what's the status on the lines above? (as quite a few are commented out)

Reply:
ok, will be changed in a followup.